jme3-alloc/src/main/java/com/jme3/alloc/util/NativeBufferUtils.java

```diff
@@ -24,77 +55,69 @@ private NativeBufferUtils() {
-     * @param value the value to set the memory blocks with starting from the buffer
-     * @param size the size to extend after the buffer address
-     * @return the buffer block
+     * @param size the size of the byte buffer in bytes units
+     * @return a newly created zeroed {@link java.nio.ByteBuffer} with the [size] bytes
+     * @see NativeBufferUtils#memoryAlloc(long)
      */
-    public static native void memorySet(final Buffer buffer, final int value, final long size); 
+    public static native ByteBuffer clearAlloc(final long size);
 
     /**
+     * Copies the value [value] into [size] byte elements starting from the [buffer] address.
      * 
-     * @param to
-     * @param from
-     * @param size
-     * @return
+     * @param buffer a {@link java.nio.ByteBuffer} to set its elements
+     * @param value an 8-bit value to set the memory blocks (buffer elements) with, starting from the buffer address
+     * @param size the number of byte elements to set
+     * @return the java-nio byte buffer for chained invocations
      */
-    public static native void memoryMove(final Buffer to, final Buffer from, final long size);
+    public static native ByteBuffer memorySet(final ByteBuffer buffer, final int value, final long size); 
 
     /**
-     * Allocates a buffer block of size = [size] bytes using {@link NativeBufferUtils#malloc(long)}, and sets its contents to zero.
+     * Manipulates the values of byte elements [size] starting from the buffers addresses, from a bytebuffer to another.
      * 
-     * @param size the size of the single memory block
-     * @return a newly created zeroed java-nio buffer with the size of [count * size]
+     * @param to the destination {@link java.nio.ByteBuffer}
+     * @param from the source {@link java.nio.ByteBuffer}
+     * @param size the number of elements to move
+     * @see NativeBufferUtils#memoryCopy(ByteBuffer, ByteBuffer, long)
      */
-    public static native Buffer clearAlloc(final long size);
+    public static native void memoryMove(final ByteBuffer to, final ByteBuffer from, final long size);
 
     /**
-     * Make a buffer block previously allocated by {@link NativeBufferUtils#malloc(long)} larger or smaller, possibly by 
-     * copying it to a new location.
+     * Copies the values of byte elements [size] starting from the buffers addresses, from a bytebuffer to another.
      * 
-     * @param buffer a java nio buffer to re-size it
-     * @param size a new size to assign to the buffer
-     * @return a newly re-sized java-nio buffer
+     * @param to the destination {@link java.nio.ByteBuffer}
+     * @param from the source {@link java.nio.ByteBuffer}
+     * @param size the number of elements to move
+     * @see NativeBufferUtils#memoryMove(ByteBuffer, ByteBuffer, long)
      */
-    public static native Buffer reAlloc(final Buffer buffer, final long size);
+    public static native void memoryCopy(final ByteBuffer to, final ByteBuffer from, final long size);
```

Comment:
Usually, from should come first and to next.

Reply:
The native method starts with the [to] buffer.